cmd/clusterctl/cmd/completion.go

```diff
@@ -0,0 +1,240 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+const completionBoilerPlate = `
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+`
+
+var (
+	completionLong = LongDesc(`
+		Output shell completion code for the specified shell (bash or zsh).
+		The shell code must be evaluated to provide interactive completion of
+		clusterctl commands. This can be done by sourcing it from the
+		.bash_profile.
+
+		Note: this requires the bash-completion framework.
+
+		To install it on Mac use homebrew:
+		    $ brew install bash-completion
+		Once installed, bash_completion must be evaluated. This can be done by
+		adding the following line to the .bash_profile
+		    $ source $(brew --prefix)/etc/bash_completion
+
+		If bash-completion is not installed on Linux, please install the
+		'bash-completion' package via your distribution's package manager.
+
+		Note for zsh users: [1] zsh completions are only supported in versions of zsh >= 5.2`)
+
+	completionExample = Examples(`
+		# Install bash completion on a Mac using homebrew
+		brew install bash-completion
+		printf "\n# Bash completion support\nsource $(brew --prefix)/etc/bash_completion\n" >> $HOME/.bash_profile
+		source $HOME/.bash_profile
+
+		# Load the clusterctl completion code for bash into the current shell
+		source <(clusterctl completion bash)
+
+		# Write bash completion code to a file and source it from .bash_profile
+		clusterctl completion bash > ~/.kube/clusterctl_completion.bash.inc
+		printf "\n# clusterctl shell completion\nsource '$HOME/.kube/clusterctl_completion.bash.inc'\n" >> $HOME/.bash_profile
+		source $HOME/.bash_profile
+
+		# Load the clusterctl completion code for zsh[1] into the current shell
+		source <(clusterctl completion zsh)`)
+
+	completionCmd = &cobra.Command{
+		Use:       "completion SHELL",
+		Short:     "Output shell completion code for the specified shell (bash or zsh)",
+		Long:      LongDesc(completionLong),
+		Example:   completionExample,
+		Args:      cobra.ExactArgs(1),
+		RunE:      runCompletion,
+		ValidArgs: GetSupportedShells(),
+	}
+)
+
+var (
```

Comment:
No need for another var section
```suggestion

```

Reply:
Fixed by https://github.com/kubernetes-sigs/cluster-api/pull/3622/commits/9ad122d707ed336e2c86d720f305d25140370e53.